### PR TITLE
replication: Add MRF backlog API

### PR DESCRIPTION
**Pull request by poornas**
Return most recent failed replication entries

### CODE REVIEW

File: examples/replicate-mrf.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
+//go:build ignore
+// +build ignore
+
+// Copyright (c) 2015-2022 MinIO, Inc.
```
Comment (harshavardhana) [inline examples/replicate-mrf.go:4]:
```suggestion
// Copyright (c) 2015-2023 MinIO, Inc.
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: replication-api.go [hunk 2]
```diff
@@ -109,3 +109,61 @@ func (adm *AdminClient) BucketReplicationDiff(ctx context.Context, bucketName st
 	// Returns the diff channel, for caller to start reading from.
 	return diffCh
 }
+
+// ReplicationMRF rrepresents MRF backlog for a bucket
```
Comment (klauspost) [inline replication-api.go:113]:
```suggestion
// ReplicationMRF represents MRF backlog for a bucket
```

File: replication-api.go [hunk 3]
```diff
@@ -109,3 +109,61 @@ func (adm *AdminClient) BucketReplicationDiff(ctx context.Context, bucketName st
...
+		if node != "" {
+			queryValues.Set("node", node)
+		}
+		reqData := requestData{
+			relPath:     adminAPIPrefix + "/replication/mrf",
+			queryValues: queryValues,
+		}
+
+		// Execute PUT on /minio/admin/v3/mrf to get mrf backlog for a bucket.
+		resp, err := adm.executeMethod(ctx, http.MethodPost, reqData)
```
Comment (klauspost) [inline replication-api.go:142]: Put or Post?

Really it should be a GET ;)

**Review by klauspost — COMMENTED** [review]
Is this an event stream of failures?

I was initially expecting something that would fit into the Metrics API, but looking more at it, it does seem different.

Are you adding a Metrics type later that aggregates replication stats?

**Comment by poornas** [comment]
> Is this an event stream of failures?
> 
> I was initially expecting something that would fit into the Metrics API, but looking more at it, it does seem different.
> 
> Are you adding a Metrics type later that aggregates replication stats?

this is an event stream of failures, will check out the Metrics API in other PR that does the replication stats.

File: api-log.go [hunk 4]
```diff
@@ -102,7 +102,7 @@ func (adm AdminClient) GetLogs(ctx context.Context, node string, lineCnt int, lo
 				queryValues: urlValues,
 			}
 			// Execute GET to call log handler
-			resp, err := adm.executeMethod(ctx, http.MethodGet, reqData)
+			resp, err := adm.executeMethod(ctx, http.MethodPost, reqData)
```
Comment (klauspost) [inline api-log.go:105]: ? Seems like an unintentional change?

File: replication-api.go [hunk 5]
```diff
@@ -109,3 +117,60 @@ func (adm *AdminClient) BucketReplicationDiff(ctx context.Context, bucketName st
 	// Returns the diff channel, for caller to start reading from.
 	return diffCh
 }
+
+// ReplicationMRF represents MRF backlog for a bucket
+type ReplicationMRF struct {
+	NodeName   string `json:"nodeName" msg:"n"`
```
Comment (vadmeste) [inline replication-api.go:123]: Shouldn't we add a Version here, so it is easy to error out when there is an incompatibility in the format and we can make madmin-go supports multiple versions. Maybe we can use major/minor version as well ?

**Review by vadmeste — CHANGES_REQUESTED** [review]
A comment

File: api-log.go (hunk 4 above)
Comment (poornas) [reply api-log.go]: thanks, removed this

File: replication-api.go (hunk 5 above)
Comment (poornas) [reply replication-api.go]: done

**Comment by poornas** [comment]
@vadmeste , @harshavardhana PTAL

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**